### DISCUSSION TITLE
v4 - useResult could return a Ref instead of a Ref<readonly>.

### DISCUSSION THREAD

**Discussion by fenilli** [Akryum/vue-apollo#1261] [Q&A] [answered]
**Is your feature request related to a problem? Please describe.**
When trying to fetch fields for a form, you need to use, a `reactive` + `useResult` to populate the `reactive`, after you leave and come back to that component, useQuery will fetch the cache, so onResult will not trigger, so you also need to before `onResult` check `if (result.value)` and populate the `reactive` property, that takes a lot of try and error when using for the first time.

just because useResult returns a readonly result, meaning that you cant use those inside forms.

**Describe the solution you'd like**
have a composable useResultRef that works the same as useResult but returns a Ref instead.

**Describe alternatives you've considered**
Tried using toRefs, but it loses its reactivity.

**Comment by Akryum** [comment]
Apollo client returns immutable data.

**Comment by fenilli** [comment, marked as answer]
So how would you go for fetching forms fields for updating? What i'm doing right now is using the result and onResult to do basically the same thing, but onResult is for the first result, and result for when there is a cached result, so using those I `Object.assing(fields, result);`

**Reply by joel-wenzel** [reply]
You should be able to just clone it in your useResult function
```
const item = useResult(result, null, data => JSON.parse(JSON.stringify(data.item)))
```
